homeassistant/components/modbus/sensor.py

```diff
@@ -121,6 +121,24 @@ async def async_setup_platform(
             del entry[CONF_REGISTER_TYPE]
 
     for entry in discovery_info[CONF_SENSORS]:
+        if CONF_HUB in entry:
+            # from old config!
+            hub = hass.data[MODBUS_DOMAIN][entry[CONF_HUB]]
+        else:
+            hub = hass.data[MODBUS_DOMAIN][discovery_info[CONF_NAME]]
+        if CONF_SCAN_INTERVAL not in entry:
+            entry[CONF_SCAN_INTERVAL] = DEFAULT_SCAN_INTERVAL
+        sensors.append(ModbusRegisterSensor(hub, entry))
+
+    async_add_entities(sensors)
+
+
+class ModbusRegisterSensor(RestoreEntity, SensorEntity):
+    """Modbus register sensor."""
+
+    @staticmethod
+    def create_data_structure(entry: dict) -> tuple[str | None, str | None]:
```

Comment:
small suggestion, we try to use _<name> for methods that are not to be used publicly.